webapp/src/js/services/resource-icons.js

```diff
@@ -8,87 +8,126 @@ angular.module('inboxServices').factory('ResourceIcons',
+            content = atob(icon.data);
+          } else {
+            // OTHER: base64 encode the img src
+            content = `<img src="data:${icon.content_type};base64,${icon.data}" />`;
+          }
+          cache[i].htmlContent[name] = content;
         }
-        cache.htmlContent[name] = content;
+        return cache[i].htmlContent[name];
+      } catch(e) {
+        return '&nbsp';
       }
-      return cache.htmlContent[name];
     };
 
-    var getHtml = function(name) {
-      var image = getHtmlContent(name);
-      return '<span class="resource-icon" title="' + name + '">' + image + '</span>';
+    const getHtml = (name, docId) => {
+      const image = getHtmlContent(name, docId);
+      return `<span class="${cssClass[docType.indexOf(docId)]}" title="${name}">${image}</span>`;
     };
 
-    var updateDom = function($elem) {
+    const updateDom = ($elem, doc) => {
       $elem = $elem || $(document.body);
-      $elem.find('.resource-icon').each(function() {
-        var $this = $(this);
-        $this.html(getHtmlContent($this.attr('title')));
+      const css = cssClass[docType.indexOf(doc)];
+      $elem.find(`.${css}`).each((i, child) => {
+        const $this = $(child);
+        $this.html(getHtmlContent($this.attr('title'), doc));
       });
+      if (document.getElementById('app') && doc === 'branding') {
+        document.getElementById('app').innerHTML = cache[doc].doc.title;
+      }
     };
 
-    var updateResources = function() {
+    const updateResources = docId => {
       return DB()
-        .get('resources', { attachments: true })
-        .then(function(res) {
-          cache = {
-            doc: res,
-            htmlContent: {}
-          };
-          updateDom();
+        .get(docId, { attachments: true })
+        .then(res => {
+          cache[docId].doc = res;
+          cache[docId].htmlContent = {};
+          updateDom($(document.body), docId);
         })
-        .catch(function(err) {
+        .catch(err => {
           if (err.status !== 404) {
             $log.error('Error updating icons', err);
           }
         });
     };
+    
+    docType.forEach(doc => {
+      updateResources(doc);
+    });
 
     Changes({
-      key: 'ResourceIcons',
-      filter: function(change) {
-        return change.id === 'resources';
-      },
-      callback: updateResources
+      key: 'resource-icons',
+      filter: change => change.id === 'resources',
+      callback: change => {
+        updateResources(change.id);
+      }
+    });
+
+    Changes({
+      key: 'branding-images',
+      filter: change => change.id === 'branding',
+      callback: change => {
+        updateResources(change.id);
+      }
     });
 
-    var init = updateResources();
+    Changes({
+      key: 'partner-images',
+      filter: change => change.id === 'partners',
+      callback: change => {
+        updateResources(change.id);
+      }
+    });
```

Comment:
Instead of having three different Changes calls you can have just one:

```js
    const DOC_IDS = ['resources', 'branding', 'partners'];
    Changes({
      key: 'resources',
      filter: change => DOC_IDS.includes(change.id),
      callback: change => updateResources(change.id)
    });
```